charts/k8s-infra/templates/_config.tpl

```diff
@@ -146,6 +194,126 @@ exporters:
 {{- $config | toYaml }}
 {{- end }}
 
+{{- define "opentelemetry-collector.applyOtlpExporterSelfTelemetryConfig" -}}
+{{- $config := mustMergeOverwrite (include "opentelemetry-collector.otlpExporterSelfTelemetryConfig" .Values | fromYaml) .config }}
+{{- $config | toYaml }}
+{{- end }}
+
+
+
+{{- define "opentelemetry-collector.otlpExporterSelfTelemetryConfig" -}}
+exporters:
+  otlphttp/own_telemetry:
+    {{- if .Values.presets.selfTelemetry.endpoint }}
+    endpoint: http{{ if not .Values.presets.selfTelemetry.insecure }}s{{ end }}://{{ .Values.presets.selfTelemetry.endpoint }}
+    tls:
+      insecure: {{ .Values.presets.selfTelemetry.insecure }}
+      insecure_skip_verify: {{ .Values.presets.selfTelemetry.insecureSkipVerify }}
+    headers:
+      "signoz-access-token": "${env:SIGNOZ_API_KEY}"
```

Comment:
I don't think the usage of `SIGNOZ_API_KEY` is for self telemetry everywhere.

You should use the `apiKey` that is being passed. This is the API Key for the infra data.